packages/query/src/block-processor.ts

```diff
@@ -388,4 +409,46 @@ export class BlockProcessor implements BlockProcessorInterface {
       await this.indexedDb.upsertValidatorInfo(validatorInfoResponse.validatorInfo);
     }
   }
+
+  private async updatePrices(swapOutputs: BatchSwapOutputData[], height: bigint) {
+    for (const swapOutput of swapOutputs) {
+      const swapAsset1 = getSwapAsset1(swapOutput);
+      const swapAsset2 = getSwapAsset2(swapOutput);
+
+      let numerairePerUnit = 0;
+      let pricedAsset: AssetId | undefined = undefined;
+
+      // case for trading pair <pricedAsset,numéraire>
+      if (swapAsset2.equals(NUMERAIRE_TOKEN_ID)) {
+        pricedAsset = swapAsset1;
+        // numerairePerUnit = lambda2/(delta1-unfilled1)
+        numerairePerUnit = calculatePrice(
+          getDelta1Amount(swapOutput),
+          getUnfilled1Amount(swapOutput),
+          getLambda2Amount(swapOutput),
+        );
+      }
+      // case for trading pair <numéraire,pricedAsset>
+      else if (swapAsset1.equals(NUMERAIRE_TOKEN_ID)) {
+        pricedAsset = swapAsset2;
+        // numerairePerUnit = lambda1/(delta2-unfilled2)
+        numerairePerUnit = calculatePrice(
+          getDelta2Amount(swapOutput),
+          getUnfilled2Amount(swapOutput),
+          getLambda1Amount(swapOutput),
+        );
+      }
+
+      if (pricedAsset === undefined || numerairePerUnit === 0) continue;
+
+      await this.indexedDb.updatePrice(pricedAsset, NUMERAIRE_TOKEN_ID, numerairePerUnit, height);
+    }
+  }
```

Comment:
I think it's worth adding some tests for this. Some ideas:

- We move updatePrices into its own file as a separate function (not on the class). This would require passing in the deps (like storage) into the function. When separated, we can write tests.

- If not, just start creating a test suite `block-processor.test.ts` passing in deps on the class constructor.